sheep/gateway.c

```diff
@@ -71,6 +71,11 @@ static void *init_erasure_buffer(struct request *req, int buf_len)
 	uint64_t tail = round_down(off + len, SD_EC_DATA_STRIPE_SIZE);
 	int ret;
 
+	buf = valloc(buf_len);
+	if(unlikely(!buf))
+		return NULL;
+	memset(buf, 0, buf_len);
```

Comment:
Thanks for your pointing, I didn't notice the zero clear in xvalloc(). As you say, keeping the zero copy would be safer.